Sync v2-7-stable with v2-7-test to release 2.7.2

ephraimbuddy opened the pull request:
Time for `2.7.2rc1`!

potiuk committed:
Improve back-reference generation (#33149)

* better command line support with better auto-complete
* allows to run back-referance generation for individual provider packages
* better typing including using Pathlib everywhere
* handling of redirection between providers - when an operator is
  moved from one provider to the other - instead of generating invalid
  "stable" link we generate back reference for all versions of the
  provider.

(cherry picked from commit 14faf196e901e0a6fdaeb7d55e6e293b7071b501)

potiuk committed:
Better fix for add-back-references (#33207)

This is a better fix for add-back-references breeze command, as
the previous fix had a bug and processed more than just providers
when providers --gen-type was selected. See #33149.

This version fixes it in a much more consistent way:

* you have to specify what you want to work on as option
* you can  select few of the provider via auto-complete
* you can also select apache-airflow and helm-chart
* you can mix and match them if you need

All the scripts and references where `--gen-type` was used
previously have been updated and documentation was changed to
reflect it

(cherry picked from commit 8c28454842c13cad1e0f2af0ec0d9a001e103aa2)

eumiro committed:
Refactor: Simplify code in breeze (#33273)

* Refactor: Simplify code in breeze

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 7c950a85b769e2c136c968bdbe103f87a2c45d3c)

eumiro committed:
Simplify conditions on len() in dev (#33562)

(cherry picked from commit 50abdcea19a909d7f048bb0ee4ac59cc0bcbb37c)

potiuk committed:
Bump min-version for docker and docker-compose (#33572)

It's been quite some time since we bumped min versions for docker
and docker-compose for Breeze and there are already a few cases where
it held us back from more efficiently using some newer docker or
docker-compose features (for example in #33547). Also docker-compose
v1 is sufficiently old an unmaintained, that it's about time to
get rid of it and ask the users to migrate to docker-compose v2.

The new min versions propose:

* Docker: 23.0.0 released 2023-02-01 (6 months ago)
* Docker Compose 2.14.0 release 2022-12-15 (8 months ago)

Docker v1 is not supported any more and the users are directed
to migrate to v2 to continue using Breeze.

(cherry picked from commit 73a37333918abe0612120d95169b9e377274810b)

eumiro committed:
Simplify conditions on len() (#33454)

(cherry picked from commit e434ae8e4ae7f7940a86c7386245fbaaf1821965)

frazar committed:
Fix broken link to the "UPDATING.md" file (#33583)

The "UPDATING.md" file has been removed in v2.3.0 with commit 5b9bd9954bbc6ed68c217761d581f3af46269af9, and therefore it does not exist on the `main` branch. Instead, link to the file's latest version (_i.e._, v2.2.5).

(cherry picked from commit 2f8caf91abcc10c06b28825c0ffe610dbbebabe1)

eumiro committed:
Replace strftime with f-strings where nicer (#33455)

(cherry picked from commit 94f70d818482de7defa03c0aff3c213ca6b83e9e)

eumiro committed:
Refactor: Improve detection of duplicates and list sorting (#33675)

(cherry picked from commit 2dbb9633240777d658031d32217255849150684b)

eumiro committed:
Do not create lists we don't need (#33519)

(cherry picked from commit 4154cc04ce9702b09e6f13d423686fdf4cb7b877)

eumiro committed:
Refactor: Remove useless str() calls (#33629)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 85acbb4ae9bc26248ca624fa4d289feccba00836)

eumiro committed:
Use str.splitlines() to split lines (#33592)

(cherry picked from commit 0e005643280883afe88416c7f3d3425dff8f02b7)

eumiro committed:
Refactor: Use random.choices (#33631)

(cherry picked from commit ba0bab0114a430ef0ac776980f7e29b34d48b726)

eumiro committed:
Refactor: path.rglob(…) does not need ** (#33669)

(cherry picked from commit 8aa22291821327111a1bc5f0647b764880d4b9e9)

eumiro committed:
Refactor: Use f-strings (#33734)

(cherry picked from commit 94c82916294791df10d30a1cb444daf5d8a34304)

eumiro committed:
Refactor: tmp_path in tests/dag_processing (#33740)

(cherry picked from commit eb93f67b2b2478dfe413778cbd4673c95b4ca773)

hussein-awala committed:
Use a single  statement with multiple contexts instead of nested  statements in core (#33769)

(cherry picked from commit 60e6847c181959d95789266cc2712cdacaf18cf9)

hussein-awala committed:
Merge multiple isintance calls for the same object in a single call (#33767)

(cherry picked from commit 5a1889a7e70114317b96fefa4b1e3f47ae15cc10)

hussein-awala committed:
Replace single quotes by double quotes in all Airflow modules (#33766)

(cherry picked from commit cfa5477aa4a987f79345c96f5fedc472f14e6f7a)

hussein-awala committed:
Replace type func by isinstance in fab_security manager (#33760)

(cherry picked from commit a44f261fcdd6b452aac0d6faaba8e134d0e6d1fe)

hussein-awala committed:
E731: replace lambda by a def method in Airflow core (#33758)

(cherry picked from commit a1d4a20548b18721aa7564a1e415ff866db2bebd)

hussein-awala committed:
remove unnecessary map and rewrite it using list in Airflow core (#33764)

(cherry picked from commit 4e545c8190d9e2a085d98c5097b7284099c7cd75)

hussein-awala committed:
Use literal dict instead of calling dict() in Airflow core (#33762)

(cherry picked from commit 1e81ed19997114cc7dc136e9fd64676a7710715a)

eumiro committed:
Refactor: Use inplace .sort() (#33743)

(cherry picked from commit 660386b5d962a77d895be34a644616a67a0e49c8)

landreasl committed:
Minor grammatical fix (#33802)

Minor grammatical fix

(cherry picked from commit 4f3d2841f6088f9619eb86f791d1fb4b375fb5bf)

hussein-awala committed:
Improve importing the modules in Airflow www package (#33810)

(cherry picked from commit b470c6bdcc801bcd57c4008a823bd768405d1736)

Bisk1 committed:
Refactor: remove unused state - SHUTDOWN (#33746)

Co-authored-by: daniel.dylag <danieldylag1990@gmail.com>
(cherry picked from commit 7faa72795185c1af4b21f207ce7e0735c4365d46)

oboki committed:
Fix typo in views (#33830)

(cherry picked from commit b68beb930fa8d84dda041e27d9fe36c2d0d72f8b)

Bisk1 committed:
Cleanup: remove remnants of 'shutdown' task instance state (#33893)

Co-authored-by: daniel.dylag <danieldylag1990@gmail.com>
(cherry picked from commit b37eaae15716ddd777c79478d175bab35a0b004c)

eumiro committed:
Replace lambdas with comprehensions (#33745)

(cherry picked from commit 9a0f54dbf9ddb4b556c66e3d24bf98a0b8c607e8)

hussein-awala committed:
Replace dict.items by dict.values when key is not used in core (#33940)

(cherry picked from commit 9d2cfb6261d26b7285b04323f216db2d8ad839e0)

hussein-awala committed:
Avoid creating unnecessary list when parsing stats datadog tags (#33943)

(cherry picked from commit c92c9611ede6b9156ebb2eaca2443340270435f8)

hussein-awala committed:
Add cancel_trigger_ids to to_cancel dequeue in batch (#33944)

(cherry picked from commit f63a94d3029485d5657232ac50efa96b2c9226a4)

hussein-awala committed:
Replace sequence concatination by unpacking in Airflow tests (#33935)

(cherry picked from commit 732eba98b2f485646646574052c1c3c57a94e07f)

hussein-awala committed:
Replace single element slice by next() in Airflow tests (#33938)

* Replace single element slice by next() in Airflow tests

* revert qubole change

* Update test_views_base.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 528459bbe932ff6318b629c618f878b280df2867)

hussein-awala committed:
Remove useless string join from core (#33969)

(cherry picked from commit b7191b8d0adae87f72bb6d1a038fb7937b6a80ae)

Lee-W committed:
docs(core-airflow): replace markdown style link with rst style link (#33990)

(cherry picked from commit 5cf92a01c069cf8345cdda86b68b9fe5e989d3c1)

hussein-awala committed:
replace loop by any when looking for a positive value in core (#33985)

(cherry picked from commit b8165941d007081f0b7b24458d461a8760909ac8)

potiuk committed:
Restore rendering of provider configuration with version_addded (#34011)

The `version_added` for packages should use package version, not
Airflow version.

Fixes: #34005
(cherry picked from commit 559507558b1dff591f549dc8b24092d900ffb0fa)

eumiro committed:
Refactor unneeded 'continue' jumps in api (#33842)

(cherry picked from commit 3b58a38e67c6645b04b8c2a1cc3bd95f32da6f12)

eumiro committed:
Refactor unneeded 'continue' jumps in utils (#33836)

(cherry picked from commit ca4cd3b2eceaaaa870dd3d3911217e0ed2060e2f)

eumiro committed:
Refactor unneeded 'continue' jumps in configuration (#33844)

(cherry picked from commit 3ae6b4e86fe807c00bd736c59df58733df2b9bf9)

pankajkoti committed:
Remove infinite animation for pinwheel, spin for 1.5s (#34020)

* Remove infinite animation for pinwheel, spin for 1.5s

* Use media query as per @ryanahamilton's suggestion

* Fix static checks

(cherry picked from commit f8a5c8bf2b23b8a5a69b00e21ff37b58559c9dd6)

pankajkoti committed:
Update CONTRIBUTING.rst information wrt Slack channels (#34025)

* Update CONTRIBUTING.rst information wrt Slack channels

Update the renamed first PR support channel and add caution
tip suggestubg to avoid cross-posting same question across
multiple channels.

* Update CONTRIBUTING.rst

(cherry picked from commit caf135f7c40ff07b31a9a026282695ac6202e6aa)

erdos2n committed:
Hide Irrelevant Dag Processor from Cluster Activity Page (#33611)

* first commit

* refactoring

* adding undefined default

* added logic to check if defined

* refactored component

* refactored not equal sign

* refactored code for boolean type instead of string

* refactored global variable. refactored conditional

* pre-commit and linting fixes

* updated jest-setup

* updated jest-setup

* precommit fix

* Update Health.tsx

replacing bottom margin with a top margin

* Update Health.tsx

removed bottom margin from triggerer

(cherry picked from commit c055e1da0b50e98820ffff8f8d10d0882f753384)

eumiro committed:
Refactor unneeded 'continue' jumps in dag processing (#33839)

(cherry picked from commit 562357528d85a1d1a3054b2f7b88511e1c5337f4)

eumiro committed:
Refactor unneeded 'continue' jumps in setup.py (#33848)

(cherry picked from commit ff1a5f2a991f267b0628a4706992029a36be8961)

hussein-awala committed:
Move the try outside the loop when this is possible in Airflow core (#33975)

* Move the try outside the loop when this is possible in Airflow core

* Use supress instead of except pass

(cherry picked from commit 8918b435be8c683bbd6bb2ffa871dbd31d476f48)

eumiro committed:
Refactor unneeded 'continue' jumps in cli (#33845)

(cherry picked from commit 05a0542b5ab1ec0f6eb86fa545bba524ad8db9f3)

potiuk committed:
Add state utils to Public Airflow Interface (#34059)

(cherry picked from commit c44e5b5ccc439fbe9a50afcb3d12c6a335518bb3)

Taragolis committed:
Remove [project] section from `pyproject.toml` (#34014)

* Add name and version of project into 'pyproject.toml'

* Remove [project] section from `pyproject.toml`

* Remove regex for update pyproject [project] version

* Fight agains static checks

(cherry picked from commit ba8ee909e4532318649df9c2d5a7ed70b357913d)

eumiro committed:
Refactor unneeded 'continue' jumps in jobs (#33846)

(cherry picked from commit e2dc0a9be1dd48c0841c2f07cb5870920d397c95)

eumiro committed:
Refactor unneeded 'continue' jumps around the repo (#33849)

(cherry picked from commit 668aace06f8e4366917878eafc20550d9129faea)

uranusjr committed:
Bring back SHUTDOWN-related constants (#34063)

This should restore code compatibility to 2.6 and prior. Note that the
constants would not appear in documentation since they are deprecated
and not use in Airflow core anyway anymore.

(cherry picked from commit 1ae29769cea047d01c22089d24772be5e56e3b50)

hussein-awala committed:
Add missing audit logs for Flask actions add, edit and delete (#34090)

(cherry picked from commit 988632fd67abc10375ad9fe2cbd8c9edccc609a5)

hussein-awala committed:
Replace sequence concatination by unpacking in Airflow core (#33934)

(cherry picked from commit 33e5d0351d7d09033c57a4a0b851b9e1b50bfb01)

Bisk1 committed:
Fix scheduler logic to plan new dag runs by ignoring manual runs (#34027)

* Fix manual task triggering scheduled tasks

Fixes #33949

* fix static checks

* static checks

* add unit test

* static check

* Undo renaming

* Update airflow/jobs/scheduler_job_runner.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

* use keyword-only arguments for last_dag_run and total_active_runs

---------

Co-authored-by: daniel.dylag <danieldylag1990@gmail.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 20d81428699db240b65f72a92183255c24e8c19b)

potiuk committed:
Reuse _run_task_session in mapped render_template_fields (#33309)

The `render_template_fields` method of mapped operator needs to use
database session object to render mapped fields, but it cannot
get the session passed by @provide_session decorator, because it is
used in derived classes and we cannot change the signature without
impacting those classes.

So far it was done by creating new session in mapped_operator, but
it has the drawback of creating an extra session while one is
already created (remnder_template_fields is always run in the
context of task run and it always has a session created already
in _run_raw_task). It also causes problems in our tests where
two opened database session accessed database at the same time
and it cases sqlite exception on concurrent access and mysql
error on running operations out of sync - likely when the same
object was modified in both sessions.

This PR changes the approach - rather than creating a new session
in the mapped_operator, we are retrieving the session from one
stored by the _run_raw_task. It is done by context manager and
adequate protection has been added to make sure that:

a) the call is made within the context manager
b) context manageer is never initialized twice in the same
   call stack

After this change, resources used by running task will be smaller,
and mapped tasks will not always open 2 DB sesions.

Fixes: #33178
(cherry picked from commit ef85c673d81cbeb60f29a978c5dc61787d61253e)

rajrao committed:
Added more example code for XCom push and pull (#34016)

* Added more example code for push and pull

Added more information on how to use push and pull.

* added code-block as per suggestion

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>

* added code-block as per suggestion

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>

* Fixed trailing white space

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>

* Update docs/apache-airflow/core-concepts/xcoms.rst

Co-authored-by: Wei Lee <weilee.rx@gmail.com>

---------

Co-authored-by: Andrey Anshin <Andrey.Anshin@taragol.is>
Co-authored-by: Wei Lee <weilee.rx@gmail.com>
(cherry picked from commit 1bd321d07885d659ce91d7c3ab08764362da0093)

hussein-awala committed:
Fix the required permissions to clear a TI from the UI (#34123)

(cherry picked from commit 2f5777c082189e6495f0fea44bb9050549c0056b)

Usiel committed:
Fix for TaskGroup toggles for duplicated labels (#34072)

* Fix for TaskGroup toggles for duplicated labels

We have been using `_.label` so far to remember which task groups are opened/closed. Unfortunately, this leads to issues when the same label is used twice, which can happen as described in the linked issue.
With this change we switch to use `_.id` instead, which must be unique (otherwise parsing the DAG would lead to `DuplicateTaskIdFound`).

Fixes #34066

* Fix toggle for Gantt as well

(cherry picked from commit b9acffa81bf61dcf0c5553942c52629c7f75ebe2)

ephraimbuddy committed:
Fix query bug in next_run_datasets_summary endpoint (#34143)

The query to get the dataset_triggered_dag_ids in the next_run_datasets_summary endpoint
errors with `str` object has no attribute dag_id. This issue is from a recent refactor on
sqlalchemy queries and the view has no unit test to detect it.
I added a fix with a unit test

(cherry picked from commit 79abcfacbec07b9e4ee374f7f4f7dc2c05b72692)

Taragolis committed:
Fix FAB-related logging format interpolation (#34139)

(cherry picked from commit fb99f6c9f89f619c69da93a030629ccab4dc52c4)

hussein-awala committed:
Fix unfound ab_user table in the CLI session (#34120)

(cherry picked from commit 2f32769bbb37236044f9c1ecb59809d82d0e3a62)

SamWheating committed:
Fix issues related to access_control={} (#34114)

* allow empty access control on dags

* Add test which demonstrates loss of information during ser/de

* Keep empty value when serializing access_control dict

(cherry picked from commit a61b5e893af83d3022aa8265f230b52c5b2ad93d)

eumiro committed:
Consolidate importing of os.path.* (#34060)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 907909329195c6655d1e2989b05609466ef50563)

eumiro committed:
Refactor: Consolidate import and usage of random (#34108)

(cherry picked from commit 4fa66d17003f10d03a13eda659bca8670bdf5052)

Usiel committed:
Ensure details panel is shown when any tab is selected (#34136)

The query param `tab` is set on the DAGs view when any of the tabs _within_ the details panel is selected, thus we should show the details panel, even if the user previously hid it.

There are 2 main scenarios where a user would navigate to such a link:
1. User clicks on one of the pills on the DAG view (Graph, Gantt, Code). Imo, the user will expect to see the relevant tab in the details panel, even if they have it hidden.
2. User clicks on a link (e.g. from DAG runs, or someone sharing a link). Less clear cut than scenario 1, but still I think the most user friendly result is to show the requested tab (in the details panel).

(cherry picked from commit 0b319e79ec97f6f4a8f8ce55119b6539138481cd)

potiuk committed:
Only tag :latest image when default python version is used (#34182)

We should only tag :latest image when default image is used because
this step always tags `apache/airflow:X.Y.Z` (defaulti Python) to become
the `apache/airflow:latest`. This step was previously run for all Python
versions which could have two effects:

* if non-default image completed before the default such tagging
  would fail as the `apache/airflow:X.Y.Z` was not ready yet to
  be used as target of `apache/airflow:latesti` tag

* if non-default image completed later, it would be no-op because
  the `apache/airflow:latest` already pointed to `apache/airflow:X.Y.Z`

(cherry picked from commit 51d8b1079b05575c1cf0ca35fcf97ec3c9e1a75b)

hussein-awala committed:
Combine similar if logics in core (#33988)

* Combine similar if logics in core

* Update airflow/models/baseoperator.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

* replace in tuple by multiple or equalities

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit d800a0de5194bb1ef3cfad44c874abafcc78efd6)

hussein-awala committed:
Move default timezone to except block (#34245)

(cherry picked from commit 7efc68c8e351c9aa1f9fa66bc67f325749396b6b)

hussein-awala committed:
Replace assert by if...raise in www package (#34249)

(cherry picked from commit 7b4fc3933bd1bbcf79acef32f00ba8981ea271b3)

hussein-awala committed:
Replace assert by if...raise in decorators package (#34250)

(cherry picked from commit 0470f3af6ddeda0917cd3a0616fa3206bad00236)

jscheffl committed:
Make param validation consistent for DAG validation and triggering (#34248)

(cherry picked from commit 3e340797ab98a06b51b2930610b0abb0ad20a750)

pierrejeambrun committed:
Add SBOM entry in security documentation (#34261)

(cherry picked from commit 18c6c3720d10eba5ff7bea40ea8661bf5d7627e4)

pankajkoti committed:
Change "not all" to "any" for ease of readability (#34259)

(cherry picked from commit 21ac3b92d2213b631c918f851974701fa5d55956)

hussein-awala committed:
Replace unnecessary dict comprehension with dict() in core (#33858)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 6a33d058a14cefe054c5aa4a683124220b53f005)

csp33 committed:
Set loglevel=DEBUG in 'Not syncing DAG-level permissions' (#34268)

(cherry picked from commit 8035aee8da3c4bb7b9c01cfdc7236ca01d658bae)

eumiro committed:
Refactor: Simplify string generation (#34118)

(cherry picked from commit a1fe77bc820a0dccf170fc123ec7e7280747e145)

eumiro committed:
Refactor: Simplify comparisons (#34181)

(cherry picked from commit 94d07908a2188eb650bfab21d89a49b287aee35c)

eumiro committed:
Refactor integer division (#34180)

(cherry picked from commit 9571d37cad376ab931dadf02304387c5286c1938)

hussein-awala committed:
replace = by is for type comparaison (#33983)

(cherry picked from commit 8b6ab5a392f816c19ed48d24621a5390059d1109)

yermalov-here committed:
Fix www cluster_activity view not loading due to standaloneDagProcessor templating (#34274)

(cherry picked from commit 3d8ea7926517cf2aa54f8daad8ea122a7f6efe9a)

Taragolis committed:
Change links to SQLAlchemy 1.4 (#34288)

(cherry picked from commit ed552e294596b27288fb040abd1634cb77e419c9)

potiuk committed:
Update min-sqlalchemy version to account for latest features used (#34293)

Some of the recent sqlalchemy changes are not working with minimum
version of sqlalchemy of ours - for example `where` syntax does
not allow moe than one clause and we are already passing more
in _do_delete_old_records (added in #33527). This syntax however
was added in SQL Alchemy 1.4.28 and our minimum version was
1.4.27.

This change bumps the minimum SQLAlchemy version to 1.4.28 but it also
adds a special test job that only runs on Postgres that downgrades
the SQLAlchemy to the minimum supported version (retrieved from
setup.cfg). This way, we will be able to detect such incompatible
changes at the PR time. This is a new flag `--downgrade-sqlalchemy`
on test command that works similar to earlier `--upgrade-boto`.

We also enable the `--upgrade-boto` and `--downgrade-sqlalchemy` flags
to be used for `breeze shell` command - thanks to that we can
easily test both flags with `breeze shell` command.

(cherry picked from commit efbead9fe7462b3634b6d9c842bd9a7ac78a0207)

potiuk committed:
Add "security infrastructure" paragraph to security model (#34301)

Making clear that Deployment Manger is responsible for setting up
the right infrastructure to apply some security expectations that
organisations might have when deploying applications on premises
or in their clouds.

(cherry picked from commit 09b8b06c54d54efde8750dcdc0983b391c01cd2b)

Abhishek-kumar-ISM committed:
correcting wrong time showing in grid view (#34179)

* correcting wrong time showing in grid view

* lint errors resolved

* lint errors resolved

(cherry picked from commit 8871f9591cc164f7692bd3fa0088f5e565d7d498)

Taragolis committed:
Add link to Amazon Provider Configuration in Core documentation (#34305)

(cherry picked from commit a81c7b7975d88272ab254bca2738fcf23f12be61)

ephraimbuddy committed:
REST API: Fix task instance access issue in the batch endpoint (#34315)

Currently, there's no restriction on the task instances a user can access in
the REST API batch fetch task instances endpoint.
This PR fixes it

(cherry picked from commit 3df1af4c45705d67598753a96debf5619bbfee04)

jscheffl committed:
Provide tabular overview about trigger form param types (#34285)

(cherry picked from commit 2e52d65639f3a07f57d33aad6cce072dc6548a96)

DjVinnii committed:
Add DagRun events (#34328)

Signed-off-by: Vincent Kling <vincentkling@msn.com>
Co-authored-by: Vincent Kling <vincent.kling@rabobank.nl>
(cherry picked from commit 35818ee715db48355070b3506caf5520fc452ba5)

aneojgurhem committed:
docs: correct typo in best-practices.rst (#34361)

(cherry picked from commit f93b046a38baa8eb4e4b513ef515635bc6c20fe8)

Taragolis committed:
Add information about drop support MsSQL as DB Backend in the future (#34375)

(cherry picked from commit a122b5709680e2d208a76bc5d40bd039b29dbd4e)

yanfangli85 committed:
Fix spelling errors in readme and license files (#34383)

(cherry picked from commit decac54722fd3fc20633541190ccb97edcc5daa9)

eumiro committed:
Refactor os.path.splitext to Path.* (#34352)

(cherry picked from commit 4869575b2c538b54cbc9368791a924f7cd5f7ce8)

TheCodyRich committed:
Update cluster-policies.rst (#34174)

Removed experimental tag from Pluggy interface. Resolves #34147

(cherry picked from commit 88623acae867c2a9d34f5030809102379080641a)

hussein-awala committed:
Fix dag warning endpoint permissions (#34355)

* Fix dag warning endpoint permissions

* update the query to have an accurate result for total entries and pagination

* add unit tests

* Update test_dag_warning_endpoint.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 3570bbfbea69e2965f91b9964ce28bc268c68129)

XD-DENG committed:
Add LocalKubernetesExecutor in the config.yml's executor description (#34414)

(cherry picked from commit e64269e63c2f772740d02eea01ea60f83f426fdc)

fritz-astronomer committed:
Docs for triggered_dataset_event (#34410)

* add templates reference for triggering_dataset_events and a note to check the templates page on the datasets page

* add working example, correct type of triggering_dataset_events

* explain | first | first

(cherry picked from commit 21610c1d6763e61108fae35585536a5c409d5cbc)

erdos2n committed:
Schedule default value description (#34291)

* added clarification around schedule parameter

* rewriting explanation of default value

* Update airflow/models/dag.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

* Update dag.py

* Fix trailing whitespace

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 5ab7517258716445ac583c41656fdb17f87f57a8)

starone99 committed:
Deprecate numeric type python version in PythonVirtualEnvOperator (#34359)

* Remove float type python version in PythonVirtualEnvOperator

Remove float type python version in PythonVirtualEnvOperator

* Remove int type python version in PythonVirtualEnvOperator

* change deprecated to removed

change deprecated to removed

* change removal to deprecation

change removal to deprecation

* fix typo

fix typo

* fix line too long

fix line too long

* change condition statement

change condition statement

* Use 'is not'

---------

Co-authored-by: kyeonghoon Kim <kyeonghoon.kim@bagelcode.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit b23d3f964b2699d4c7f579e22d50fabc9049d1b6)

hussein-awala committed:
Clarify var.value.get  and var.json.get usage (#34411)

(cherry picked from commit 03db0f6b785a4983c09d6eec7433cf28f7759610)

Taragolis committed:
Fix class reference in Public Interface documentation (#34454)

(cherry picked from commit 26f6a51c47804d28779fa6277b48497c113fe721)

yiannis-had committed:
Change two whitespaces to one (#34519)

(cherry picked from commit a1bd8719581f2ef1fb25aeaa89e3520e8bc81172)

shubhamraj-git committed:
using seconds for failed scenarios too (#34532)

Co-authored-by: Shubham <shubhamraj@cloudera.com>
(cherry picked from commit 117e40490865f04aed38a18724fc88a8cf94aacc)

sa1 committed:
Add more exemptions to lengthy metric list (#34531)

Co-authored-by: Saurabh Kumar <mail@sa1.me>
(cherry picked from commit fa6ca5d5316a9bd759a702e1688a69b19e4e63bc)

potiuk committed:
Support rootless mode for docker. (#34537)

In case docker is run in rootless mode, the host UID is mapped to root
user automatically and host user id is mapped to 100999 (unknown) so
changing ownership for created files in rootless mode is problematic
as it makes the generated files inaccessible

(cherry picked from commit 0631af86525ad98e90cdc0bf120df7192ea2e912)

potiuk committed:
Avoid WSL2 ones when finding a context for Breeze (#34538)

* Avoid WSL2 ones when finding a context for Breeze

* fixup! Avoid WSL2 ones when finding a context for Breeze

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit 2e764fb0978fde33f59918416bd2732294c4bf23)

pankajastro committed:
Restore EXISTING_ROLES from security.py (#34523)

(cherry picked from commit 5d2cea4515cd656e85c39e04d148fcb2d6ba516a)

jedcunningham committed:
Fix screenshot in dynamic task mapping docs (#34566)

(cherry picked from commit d2be036ace49f8e5dd98780c75a37106544bf4d6)

Taragolis committed:
Fix ODBC Connection page formatting (#34572)

(cherry picked from commit af03051962abf222cc8f05b243451b9675d7ee00)

janfrederik committed:
Fix: Add 3.11 as supported Python version (#34575)

* Fix: Add 3.11 as supported Python version

On the Quick Start page
Fixes #34574

* Update docs/apache-airflow/start.rst

---------

Co-authored-by: Hussein Awala <hussein@awala.fr>
(cherry picked from commit 9b96f76ac820b3dc020286b685a236da842e407c)

hussein-awala committed:
Avoid top-level airflow import to avoid circular dependency (#34586)

(cherry picked from commit d1b7bca6a36c146119fb5746019116c4d1e15275)

hussein-awala committed:
Fix is_parent_mapped value by checking if any of the parent tg is mapped (#34587)

(cherry picked from commit 97916ba45ccf73185a5fbf50270a493369da0344)

Lee-W committed:
fix(cli): remove "to backfill" from --task-regex help message (#34598)

this arg is used by both "airflow tasks clear" and "airflow tasks backfill"
and it does not make sense for "airflow tasks clear" to have the description "to backfile"

(cherry picked from commit c019cf18dd1be3b20baf7503326a53002c236b45)

dstandish committed:
Clarify what landing time means in doc (#34608)

(cherry picked from commit f99e65b4f3df71baf2fdf738643bbda1263e15a0)

ferruzzi committed:
Fix some whitespace (#34632)

(cherry picked from commit 97de019995185cba1e7e63ea525d099ff5c94ea7)

uranusjr committed:
Use iterative loop to look for mapped parent (#34622)

(cherry picked from commit d9ba152c15dd50baa1fef41a63424225ba8ddd47)

jedcunningham committed:
Don't run isort when creating new alembic migrations (#34636)

(cherry picked from commit db89a33b60f46975850e3f696a7e05e61839befc)

Jayden-Chiu committed:
fix connections exported output (#34640)

(cherry picked from commit a5f5e2fc7f7b7f461458645c8826f015c1fa8d78)

dstandish committed:
Fail dag test if defer without triggerer (#34619)

If user runs dag.test and task defers and no triggerer is running, we should fail so user does not sit there waiting forever.

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit e81bb487796780705f6df984fbfed04f555943d7)

Taragolis committed:
Restrict `astroid` version < 3 (#34658)

(cherry picked from commit 19450e03f534f63399bf5db2df7690fdd47b09c8)

hussein-awala committed:
Fix non deterministic datetime deserialization (#34492)

tzname() does not return full timezones and returned short hand notations are not deterministic. This changes the serialization to be deterministic and adds some logic to deal with serialized short-hand US Timezones and CEST.

---------

Co-authored-by: bolkedebruin <bolkedebruin@users.noreply.github.com>
(cherry picked from commit a3c06c02e31cc77b2c19554892b72ed91b8387de)

bolkedebruin committed:
Document how to use the system's timezone database (#34667)

The latest release of pendulum (2.1.2) contains an outdated
timezone database. It is better to rely on the local
system's database.

(cherry picked from commit d0f246398ff871bfa177f91912980d8a0f0f1c50)

jamesqo committed:
Correct docs for multi-value select (#34690)

(cherry picked from commit 8e935861f6bd0529543756d3b750de218aef5462)

Calder-Ty committed:
Fix: make dry run optional for patch task instance  (#34568)

* fix: Make dry_run optional per docs

This fixes an issue where dry_run is not actually and optional parameter
in the patch task_instance api.

* chore: remove formatting changes

* fix: Make changes for api docs

This updates the docs and the code so that they are in alignment while
also being consistent with all other endpoints. All other Endpoints have
dry run set to be True by default.

* fix: Update static ts file for api change

* fix: Remove dump_default

(cherry picked from commit a4357ca25cc3d014e50968bac7858f533e6421e4)

dstandish committed:
Fix foreign key warning re ab_user.id (#34656)

Introduced in #34120.

(cherry picked from commit 1fdc2311250fbae47749822b192a99066600f8ad)

hussein-awala committed:
Fix broken breeze by fixing package version (#34701)

(cherry picked from commit 6618c5f90d037d57e9f3bf1e90cd0712426d6caa)

Commit:
Fix SesssionExemptMixin spelling (#34696)

Co-authored-by: David Kalamarides <david.kalamarides@capitalone.com>
(cherry picked from commit 63945c71241e7b1b278068e1786e610facd569e0)

alartum committed:
Fixed rows count in the migration script (#34348)

* Fixed row count for SQLAlchemy 1.4+

* Updated newsfragments

* Fixed typo

* Added newline

* Added test for `check_bad_references`

(cherry picked from commit f349fda125c2251ac4129c2c28fbf6f7dbb69294)

uranusjr committed:
Parse 'docker context ls --format=json' correctly (#34711)

(cherry picked from commit 19284981f88e45dca4c4003837e3cead1723caf1)

ephraimbuddy committed:
Order cherrypicking candidates by the closed date (#34739)

Currently, the candidates are ordered by the PR number but ordering by the closed date
is better and will help avoid some conflicts

(cherry picked from commit 90628e480a61e7b4991d42a536b79e634dee7c12)

hussein-awala committed:
Import AUTH_REMOTE_USER from FAB in WSGI middleware example (#34721)

(cherry picked from commit feaa5087e6a6b89d9d3ac7eaf9872d5b626bf1ce)

ephraimbuddy committed:
Update version to 2.7.2

ephraimbuddy committed:
Update RELEASE_NOTES.rst

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,117 @@
+- Add missing audit logs for Flask actions add, edit and delete (#34090)
+- Hide Irrelevant Dag Processor from Cluster Activity Page (#33611)
+- Remove infinite animation for pinwheel, spin for 1.5s (#34020)
+- Restore rendering of provider configuration with ``version_added`` (#34011)
```
this is internal change is it not?

ephraimbuddy replied:
It's a bug: https://github.com/apache/airflow/issues/34005

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,117 @@
+
+Bug Fixes
+"""""""""
+- Import ``AUTH_REMOTE_USER`` from FAB in WSGI middleware example (#34721)
```
this is doc only change

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,117 @@
+Bug Fixes
+"""""""""
+- Import ``AUTH_REMOTE_USER`` from FAB in WSGI middleware example (#34721)
+- Fixed rows count in the migration script (#34348)
```
We can replace this with a more informative note user added in the newsfragment

`Fixed AttributeError: 'Select' object has no attribute 'count' during the airflow db migrate command`

utkarsharma2 committed:
Fix autodetect_docker_context for list of dict case (#34779)

* Fix autodetect_docker_context for list of dict case

* Handle the case of dict

(cherry picked from commit 5c2dc53bcb17ae515f9565c41d15cc6d1693382c)

bolkedebruin committed:
Add support for ZoneInfo and generic UTC (#34683)

* Add support for ZoneInfo and generic UTC

Certain providers rely on other datetime implementations
and fail to serialize.

(cherry picked from commit 7707f4a9310e032476f392d25912b653771bbda2)

bolkedebruin committed:
zoneinfo.ZoneInfo is not available on < Python 3.9 (#34804)

(cherry picked from commit 25cd12d307be6359c844f4764b6f90e954bccb56)

hussein-awala committed:
Check if the lower of provided values are sensitives in config endpoint (#34712)

* Check if the lower of provided values are sensitives in config endpoint

* update unit test

* ensure that all values in sensitive dict are in lower characters

(cherry picked from commit f044589b685855a8fce8f5376bea2564c5a001f7)

potiuk committed:
Add explanation for Audit log users (#34814)

(cherry picked from commit 4b867800c573588f70c0eaa3f0bc78d434155d6c)

pierrejeambrun committed:
Clarify audit log permissions (#34815)

(cherry picked from commit 1f3525fd93554e66f6c3f2d965a0dbf6dcd82724)
